src/group.h

```diff
@@ -64,12 +64,13 @@ static void secp256k1_ge_set_gej_var(secp256k1_ge *r, secp256k1_gej *a);
 /** Set a batch of group elements equal to the inputs given in jacobian coordinates */
 static void secp256k1_ge_set_all_gej_var(secp256k1_ge *r, const secp256k1_gej *a, size_t len);
 
-/** Bring a batch inputs given in jacobian coordinates (with known z-ratios) to
+/** Bring a batch inputs given in coordinates (with known z-ratios) to
  *  the same global z "denominator". zr must contain the known z-ratios such
- *  that mul(a[i].z, zr[i+1]) == a[i+1].z. zr[0] is ignored. The x and y
- *  coordinates of the result are stored in r, the common z coordinate is
- *  stored in globalz. */
-static void secp256k1_ge_globalz_set_table_gej(size_t len, secp256k1_ge *r, secp256k1_fe *globalz, const secp256k1_gej *a, const secp256k1_fe *zr);
+ *  that mul(a[i].z, zr[i+1]) == a[i+1].z. zr[0] is ignored.
+ *  The common z coordinate is the (implicit) z coordinate of a[len-2], and therefore
+ *  the coordinates of a[len-1] are not changed.
+ */
+static void secp256k1_ge_globalz_fixup_table(size_t len, secp256k1_ge *a, const secp256k1_fe *zr);
```

Comment:
Is it also worth mentioning that `a[len - 1]` implicitly has `z = 1` (which is why it's enough for this function to accept a `_ge` rather than `_gej`, without even taking in a `Z` value)?

Reply:
This isn't correct.  The implicit z value is the "global Z" value returned in `secp256k1_ecmult_odd_multiples_table`.

Reply:
I'm probably using "implicit" incorrectly, I was taking into account this comment:

`Each point in 'prej' has a z coordinate too small by a factor of 'd.z'.`

https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_impl.h#L93

Reply:
The way I think of it is that a is an array of points with unknown z-coordinates, but known ratios of these z-coordinates.  But that is good enough to rescale all the points so that they end up all sharing the same z-coordinate of the last entry, whatever it is.